app/Http/Controllers/API/EventController.php

```diff
@@ -319,29 +354,416 @@ private function getUser() {
+     *                ),
+     *                @OA\Property(
+     *                   property="title",
+     *                   ref="#/components/schemas/Event/properties/title",
+     *                ),
+     *                @OA\Property(
+     *                   property="description",
+     *                   ref="#/components/schemas/Event/properties/description",
+     *                ),
+     *                @OA\Property(
+     *                   property="location",
+     *                   ref="#/components/schemas/Event/properties/location",
+     *                ),
+     *                @OA\Property(
+     *                   property="online",
+     *                   ref="#/components/schemas/Event/properties/online",
+     *                ),
+     *             )
+     *         )
+     *    ),
+     *    @OA\Response(
+     *        response=200,
+     *        description="Successful operation",
+     *        @OA\JsonContent(
+     *            @OA\Property(
+     *              property="data",
+     *              title="data",
+     *              ref="#/components/schemas/Event"
+     *            )
+     *        ),
+     *     )
+     *  )
+     */
+    public function updateEventv2(Request $request, $idEvents) {
+        $user = $this->getUser();
+
+        list($groupid, $start, $end, $title, $description, $location, $timezone, $latitude, $longitude, $online, $link) = $this->validateEventParams(
+            $request,
+            false
+        );
+
+        // Convert the timezone to UTC, because the timezone is not itself stored in the DB.
+        $event_start_utc = Carbon::parse($start)->setTimezone('UTC')->toIso8601String();
+        $event_end_utc = Carbon::parse($end)->setTimezone('UTC')->toIso8601String();
+        $hours = Carbon::parse($event_start_utc)->diffInHours(Carbon::parse($event_end_utc));
+
+        // We don't let the user change the group that an event is on.
+        $update = [
+            'event_start_utc' => $event_start_utc,
+            'event_end_utc' => $event_end_utc,
+            'hours' => $hours,
+            'free_text' => $description,
+            'online' => $online,
+            'venue' => $title,
+            'link' => $link,
+            'location' => $location,
+            'latitude' => $latitude,
+            'longitude' => $longitude,
+            'timezone' => $timezone
+        ];
+
+        $party = Party::findOrFail($idEvents);
+        $party->update($update);
+
+        // REVIEW: In the old code we wouldn't generate EditEvent in the approval case.  But I think that was a bug.
+        if ($request->has('moderate') && $request->input('moderate') == 'approve') {
+            $party->approve();
+        }
+
+        event(new EditEvent($party, $update));
+
+        if ($request->has('users')) {
+            // REVIEW Existing logic.  But is this needed?  If so we should put it in the OpenAPI spec.  But I
```

Comment:
I think you're right, I certainly have no recollection of us ever passing a list of users.  As far as I know we add/remove users to/from an event one by one when people perform those actions - right?  In which case, this isn't needed.

Reply:
That's what I think.  Have retired this method.